### PR TITLE
fix: set "rel" attr if "target" attr is set

### PR DESCRIPTION
**Pull request by lesterchoi-okta**
## PR Checklist

- [ ] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-810914](https://oktainc.atlassian.net/browse/OKTA-810914)

### CODE REVIEW

File: src/v2/view-builder/components/Link.js [hunk 1]
```diff
@@ -21,11 +21,14 @@ export default View.extend({
...
     let href = this.options.href || '#';
+    const rel = (this.options.target && !this.options.rel)
+      ? 'noopener noreferrer'
+      : undefined;
     return {
       'data-se': this.options.name,
       href: href,
       target: this.options.target,
-      rel: this.options.rel
+      rel: rel
```
Comment (lesterchoi-okta) [inline src/v2/view-builder/components/Link.js:31]: @glenfannin-okta in response to your comment/concern on https://github.com/okta/okta-signin-widget/pull/3721

File: test/testcafe/spec/WidgetCustomization_spec.js [hunk 2]
```diff
@@ -99,19 +99,23 @@ test.requestHooks(identifyMock)('should show custom footer links', async t => {
...
       ]
     },
   });
   await checkA11y(t);
   await t.expect(identityPage.getCustomForgotPasswordLinkUrl()).eql('https://okta.okta.com/signin/forgot-password');
   await t.expect(identityPage.getHelpLinkUrl()).eql('https://google.com');
   await t.expect(identityPage.getCustomHelpLinkUrl(0, 'What is Okta?')).eql('https://acme.com/what-is-okta');
-  await t.expect(identityPage.getCustomHelpLinkUrl(1, 'Acme Portal')).eql('https://acme.com');
   await t.expect(identityPage.getCustomHelpLinkLabel(0, 'What is Okta?')).eql('What is Okta?');
+  await t.expect(identityPage.getCustomHelpLinkRel(0, 'What is Okta?')).eql(null);
```
Comment (lesterchoi-okta) [inline test/testcafe/spec/WidgetCustomization_spec.js:113]: expecting this to fail, should be
```js
  await t.expect(identityPage.getCustomHelpLinkRel(0, 'What is Okta?')).eql('noopener noreferrer');
```

**Comment by lesterchoi-okta** [comment]
some snapshots may also fail. posting this for consideration and seeing how it does in CI

File: .testcaferc.js [hunk 3]
```diff
@@ -95,6 +96,13 @@ const config = {
     fixture: {
       before: async () => {
         await waitOn({ resources: ['http-get://localhost:3000'] });
+      },
+    },
+    test: {
+      after: async (t) => {
+        const withRel = Selector('a[target][rel]', { timeout: 10 }).with({ boundTestRun: t });
+        const withTarget = Selector('a[target]', { timeout: 10 }).with({ boundTestRun: t });
+        await t.expect(withTarget.count).eql(await withRel.count, 'anchors with "target" attr must have "rel" attr');
```
Comment (lesterchoi-okta) [inline .testcaferc.js:105]: Revertable

File: src/v2/view-builder/components/Link.js [hunk 4]
```diff
@@ -21,10 +21,14 @@ export default View.extend({
 
   attributes() {
     let href = this.options.href || '#';
+    const rel = (this.options.target && !this.options.rel)
+      ? 'noopener noreferrer'
+      : this.options.rel;
```
Comment (lesterchoi-okta) [inline src/v2/view-builder/components/Link.js:26]: Focus here

Comment (trevoring-okta) [reply src/v2/view-builder/components/Link.js:26]: So this approach would default `rel` to `"noopener noreferrer"` even when `target` isn't `"_blank"`? I assume there's no drawbacks to that?

Comment (lesterchoi-okta) [reply src/v2/view-builder/components/Link.js:26]: I'll limit it to `target="_blank"`

**Comment by trevoring-okta** [comment]
@lesterchoi-okta I can see you updated the gen3 `Link` component but we also need to make the equivalent change to actually pass `target="_blank"` to the help link in gen3 as well. You can add that here: https://github.com/okta/okta-signin-widget/blob/8c660a4277d5113e81efe9c9a70cdeaedde32955/src/v3/src/transformer/button/transformHelpLinks.ts#L34-L42

File: src/v3/src/transformer/button/transformHelpLinks.ts [hunk 5]
```diff
@@ -24,42 +24,43 @@ export const transformHelpLinks: TransformStepFnWithOptions = ({
...
+      contentType: 'footer',
+      options: {
+        href,
+        target,
+        step: '',
+        label: text,
+        dataSe: 'custom',
+      },
+    });
+    const customHelpLinks = getCustomHelpLinks(widgetProps).map(transformCustomLink);
```
Comment (trevoring-okta) [inline src/v3/src/transformer/button/transformHelpLinks.ts:57]: Is the linter fine w/ all of these changes in indentation? I assume these are coming from your local indentation preferences or something?

Comment (lesterchoi-okta) [reply src/v3/src/transformer/button/transformHelpLinks.ts:57]: no lint doesn't like it. not sure what config is kicking in to apply that change. fixing now

**Pull request merged**